.circleci/config.yml

```diff
@@ -154,7 +157,7 @@ jobs:
       - checkout
       - restore_cache:
           keys:
-            - v1-dependencies-{{ .Revision }}
+            - v1-dependencies-python:3.11-{{ .Revision }}
```

Comment:
Maybe it's a silly issue, but shouldn't we ensure that it lints correctly also in the previous Python version ?

Reply:
This might become tricky; what do we do in case of a conflict, e.g. if python3.9 linter fails - we fix the issue, but then python3.10 linter fails because of the fix?

Reply:
Then should we just lint on the oldest version, aka python3.9?

Reply:
I find it convenient when the CI linter version matches the development image version as it might avoid conflicts - should we use the oldest version in development too? Reverting most changes from #202? Or maybe make the development image configurable with a make file variable setting the python version, something like `make bootstrap PYTHON_VERSION=3.9`?

Reply:
> I find it convenient when the CI linter version matches the development image version as it might avoid conflicts - should we use the oldest version in development too? Reverting most changes from #202? Or maybe make the development image configurable with a make file variable setting the python version, something like `make bootstrap PYTHON_VERSION=3.9`?

@SergioSim I did not have this in mind. In this case, we should keep the linter with Python 3.11 version

Reply:
I think we could be opinionated regarding the python version used in quality controls (linters, etc). Only tests should reflect supported python versions.